src/test/java/org/opensearch/security/dlic/rest/api/UserApiTest.java

```diff
@@ -144,6 +148,55 @@ public void testParallelPutRequests() throws Exception {
         deleteUser("test1");
     }
 
+    @Test
+    public void testUserFilters() throws Exception {
+        setup();
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+        final int SERVICE_ACCOUNTS_IN_SETTINGS = 1;
+        final int INTERNAL_ACCOUNTS_IN_SETTINGS = 19;
+        final String serviceAccountName = "JohnDoeService";
+        HttpResponse response;
+
+        response = rh.executeGetRequest(ENDPOINT + "/internalusers?filterBy=internal");
+
+        Assert.assertEquals(response.getBody(), HttpStatus.SC_OK, response.getStatusCode());
+        JsonNode list = DefaultObjectMapper.readTree(response.getBody());
+        Assert.assertEquals(INTERNAL_ACCOUNTS_IN_SETTINGS, list.size());
+
+        response = rh.executeGetRequest(ENDPOINT + "/internalusers?filterBy=service");
+        Assert.assertEquals(response.getBody(), HttpStatus.SC_OK, response.getStatusCode());
+        list = DefaultObjectMapper.readTree(response.getBody());
+        assertThat(list, Matchers.emptyIterable());
```

Comment:
Shouldn't this be 1 (SERVICE_ACCOUNTS_IN_SETTINGS)?

Reply:
That assert being made before the service account entry is created. 

It was suggested at some point to check the assertion  befo.re and after creating the entry . Line 185 has the `        Assert.assertEquals(SERVICE_ACCOUNTS_IN_SETTINGS, list.size());`